### PR TITLE
feat(developer): debugger uses Keyman Core 🐞

### PR DESCRIPTION
**Pull request by mcdurdin**
Start of the refactor of the debugger to use Keyman Core. Basic events work but there is a lot to tidy up yet.

Some items to complete:
- [x] Single-step mode
- [x] Breakpoints
- [x] Debug status windows
- [x] Refactor debugging code out of frmDebug so we can share it for the visual keyboard import tool and command line regression testing tools
- [x] Setup appropriate path for kmnkbp0.dll

For follow-up PRs:
- [ ] Keyboard options
- [ ] Regression testing
- [ ] User-selected base layout
- [ ] Test mode does not work -- do we even need to keep it?
- [ ] Refactor ForceKeyboard property in UfrmDebug.pas
- [ ] Ensure Action_Char and Action_Deadkey have same method of inserting text
- [ ] Ensure insertion point position is tracked correctly during single-step debugging
- [ ] Define behaviour around selection (in DebugEvent.pas)
- [ ] Hide Debug Events form
- [ ] `// TODO: remove +1 -1 messes for deadkey codes //GetDeadkeyMarker(i);` in UfrmDebug.pas
- [ ] `// TODO: this method of inserting chars differs from Action_Char;` in UfrmDebug.pas

This is a big PR. I am happy for it to be reviewed in stages. It is probably best to treat UfrmDebug.pas as an entirely new file because it is essentially a rewrite. There are a number of other files which need further refactoring (all the files without namespaced filenames, essentially) which I won't try and do in this PR -- it's huge enough as it is.

Can also do a peer code review if you prefer.

### CODE REVIEW

File: windows/src/developer/TIKE/debug/Keyman.System.Debug.DebugEvent.pas [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+  end;
+end;
+
+procedure TDebugEventList.Action_EmitKeystroke(const key: Word);
+var
+  event: TDebugEvent;
+begin
+  case key of
+    VK_TAB:    Action_Char(9);  // Emit a tab character
+    VK_RETURN: Action_Char(13); // New line character
```
Comment (rc-swag) [inline windows/src/developer/TIKE/debug/Keyman.System.Debug.DebugEvent.pas:160]: Magic numbers but at least the comment tells save you looking it up

**Review by rc-swag — COMMENTED** [review]
Just a few comments otherwise looks ready to go.

File: windows/src/developer/TIKE/debug/Keyman.System.Debug.DebugEvent.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/debug/Keyman.System.Debug.DebugEvent.pas:160]: Inclined to let these ones slide given they are chars and not virtual keys (Delphi has no `\t`, instead it uses `#9`, so here I could do `Ord(#9)` for Tab but it makes no difference).

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.95-alpha](https://keyman.com/downloads/releases/alpha/15.0.95)

**Pull request merged**